tmt/base.py

```diff
@@ -115,33 +115,32 @@ class ValidateFmfMixin:
     method to perform the validation.
     """
 
-    def _validate_fmf_node(self, node: fmf.Tree, logger: tmt.utils.Common) -> None:
+    def _validate_fmf_node(self, node: fmf.Tree, logger: tmt.utils.Common,
+                           raise_on_validation_error: bool) -> None:
         """ Validate a given fmf node """
 
         errors = tmt.utils.validate_fmf_node(
             node, f'{self.__class__.__name__.lower()}.yaml')
 
-        for _, error_message in errors:
-            # TODO: how to proceed requires more context. At the beginning, log the
-            # error and continue - requires a logger, and with what level? - but
-            # later, validation would become mandatory and this would become an
-            # exception.
-            #
-            # Or will it? Wouldn't an option to ignore this kind of error be useful?
-            # We'll see, in any case, more parameters will be needed.
+        if errors:
+            if raise_on_validation_error:
+                # Raise the very first issue found - there may be more, of course...
+                raise tmt.utils.SpecificationError(errors[0][1])
```

Comment:
Could you store whole error? (`original` keyword) - I'm afraid we will need to report all issues for the user, not only the first one.

Reply:
`errors` is actually a list of errors, `[(jsonschema.ValidationError, slightly_better_error_message), ...]`. There is no single "original" exception I could attach, so, what about extending `SpecificationError` to accept, besides the `original: Exception` also this list of (original jsonschema error, slightly better error message) pairs as yet another parameter?

Reply:
+1 to extending SpecificationError

Reply:
Done.

Reply:
seems resolved